Azure/CanadaPubSecALZ
# Enable Diagnostic Settings for Log Analytics Workspace

## Overview/Summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

1. Fixes #107 

### Breaking Changes

None

## Testing Evidence

1. Deployed new policy & policy set (initiative) via Azure DevOps Pipeline.
2. Wait for policy evaluation to complete.  The log analytics workspace is now marked non-compliant because diagnostic settings are not set.

![image](https://user-images.githubusercontent.com/16688027/144293397-435549c5-02db-4da1-8abd-fa06bfdfba89.png)

![image](https://user-images.githubusercontent.com/16688027/144293324-49b90253-43b7-4da2-ad52-a689192d62b2.png)

3. Remediate the non-compliant resource through Azure Policy

![image](https://user-images.githubusercontent.com/16688027/144293557-52f168a5-5111-484e-a429-090388dd96df.png)

4. Wait for remediation to complete

![image](https://user-images.githubusercontent.com/16688027/144293662-61f62aec-5d26-49d5-9ff1-ec13da3fe0af.png)

5. Verify diagnostic settings are enabled on Log Analytics Workspace

![image](https://user-images.githubusercontent.com/16688027/144293794-fa18d688-c5b7-44b0-88e0-7abd9731cc0e.png)

![image](https://user-images.githubusercontent.com/16688027/144293867-d4d68e16-7942-4591-827c-c0591178524e.png)

![image](https://user-images.githubusercontent.com/16688027/144293904-132d6760-8460-4e0a-92c4-eded5c5e2f9d.png)

6. Verify logs populated in Log Analytics Workspace in `LAQueryLogs` table

![image](https://user-images.githubusercontent.com/16688027/144294971-887a6345-b486-4947-9279-6e0ce639eb78.png)


## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/azure/CanadaPubSecALZ/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/azure/CanadaPubSecALZ/issues)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/azure/CanadaPubSecALZ/tree/main)
